src/drivers/gps/gps.cpp

```diff
@@ -321,7 +321,7 @@ GPS::GPS(const char *path, gps_driver_mode_t mode, GPSHelper::Interface interfac
 		set_device_bus_type(device::Device::DeviceBusType::DeviceBusType_SERIAL);
 
 		char c = _port[strlen(_port) - 1]; // last digit of path (eg /dev/ttyS2)
-		set_device_bus(atoi(&c));
+		set_device_bus(c - 48); // sub 48 to convert char to integer
```

Comment:
What's wrong with atoi?

Reply:
The parameter of `atoi()` should be `char*`, which means a string. However, in C, string should end with `\0`. Use the address of a arbitrary `char` variable can't promise it ends with `\0`, hence, instead of using `atoi()` subtract 48 is a simpler and safer solution.

Reply:
Moreover, `atoi()` somehow is a dangerous function mentioned by `man`